### ISSUE TITLE
Does not catch Duplicated State names  

### ISSUE THREAD

**Issue by techmag** [awslabs/statelint#39]
A duplicate state name gets past both this linter and the initial console editor. 

It doesn't show as an error until you try to save in the console. 

While it's a relatively trivial matter it should nonetheless be easy to catch? 

More useful the longer the function is of course. 

The following passes...

```json
{
  "Comment": "Duplicate Example",
  "StartAt": "Dup Check",
  "States": {
    "Dup Check": {
      "Type": "Pass",
      "Result" : { "none" : 0 },
      "Next": "SuccessState"
    },
    "Dup Check": {
      "Type": "Pass",
      "Result" : { "none" : 0 },
      "Next": "SuccessState"
    },
    "SuccessState": {
      "Type": "Succeed"
    }
  }
}
```

**Comment by wong-a** [comment]
This is a bug. Duplicate state names aren't allowed.

From the Amazon States Language spec:
> States are represented as fields of the top-level “States” object. The state name, whose length MUST BE less than or equal to 128 Unicode characters, is the field name; state names MUST be unique within the scope of the whole state machine.
